### PR TITLE
Refactor LICENSE to include Z3-Noodler licence notification

### PR DESCRIPTION
**Pull request by Adda0**
This PR refactors the MIT licence file to include the Z3-Noodler licence notification.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,27 @@
+Z3-Noodler
+
+This project is an SMT string solver, a modified version of Z3 SMT Solver, distributed under the MIT License.
+Original Z3 repository: https://github.com/Z3Prover/z3
+
+MIT License
+
+Copyright (c) Microsoft Corporation
```
Comment (jurajsic) [inline LICENSE:8]: Is this correct? It seems like noodler is copyrighted under microsoft

**Comment by jurajsic** [comment]
DIdn't we want to keep both Z3 and add our licence?

File: LICENSE (hunk 1 above)
Comment (Adda0) [reply LICENSE:8]: Both approaches are valid. The sequence of copyrights represents the derivative work which Z3-Noodler is. The original Z3 is copyrighted under Microsoft. Therefore, the original copyright needs to remain there, too.

**Comment by Adda0** [comment]
If you prefer, I am OK with using the other approach with two LICENSE files.

File: LICENSE (hunk 1 above)
Comment (jurajsic) [reply LICENSE:8]: Ok, but we should make it more obvious, for me it looks now that microsoft is endorsing noodler development. Maybe say that Z3 is copyrighted under microsoft, and noodler is copyrighted by us?

Comment (jurajsic) [reply LICENSE]: ah ok, I didn't know that

**Review by vhavlena — APPROVED** [review]

**Review by kilohsakul — APPROVED** [review]

**Pull request merged**